### PR TITLE
Add GHC 8.4.4, update Cabal to 2.4.

### PR DESCRIPTION
**Pull request by psftw**
Needs some additional testing before merge.

Known issues:
- No upstream stackage snapshot available yet for 8.6.1
- Issue in cabal-install-2.4, but simple workaround added ( https://github.com/haskell/cabal/issues/5516 )

### CODE REVIEW

**Comment by hvr** [comment]
I'd wait a little bit longer, because

 1. GHC 8.4.4 has just been released (it's already on http://downloads.haskell.org/debian)
 2. GHC 8.6.1 is considered broken; GHC 8.6.2 is expected soon
 3. cabal 2.4.0.1 is expected shortly after GHC 8.6.2 which fixes issues such as #5516

File: 8.6/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ENV LANG            C.UTF-8
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git && \
+    echo 'deb http://downloads.haskell.org/debian stretch main' > /etc/apt/sources.list.d/ghc.list && \
+    apt-key adv --keyserver keyserver.ubuntu.com --recv-keys BA3CBA3FFE22B574 && \
+    apt-get update && \
+    apt-get install -y --no-install-recommends ghc-8.6.1 cabal-install-2.4 \
+        zlib1g-dev libtinfo-dev libsqlite3-dev g++ netbase xz-utils make && \
+    curl -fSL https://github.com/commercialhaskell/stack/releases/download/v1.7.1/stack-1.7.1-linux-x86_64.tar.gz -o stack.tar.gz && \
```
Comment (d12frosted) [inline 8.6/Dockerfile]: You can change that to `1.9.1`.

**Comment by psftw** [comment]
Merging the 8.4.4 update, skipping 8.6.1, and leaving 8.6.2 for another day.

**Comment by scott-fleischman** [comment]
@psftw Would you upload the 8.4.4 image to docker hub? Thanks!

**Comment by sgraf812** [comment]
Now that 8.6.2 is out, could you please add the image for that?

**Pull request merged**